llvm/lib/Target/X86/X86ISelLowering.cpp

```diff
@@ -30976,6 +30976,14 @@ bool X86TargetLowering::needsCmpXchgNb(Type *MemType) const {
   return false;
 }
 
+TargetLoweringBase::AtomicExpansionKind
```

Comment:
I would like to get rid of this hook. Directly promoting atomic_load to int in codegen, like any other operation, shouldn't be difficult